### ISSUE TITLE
False positive of "reportIncompatibleMethodOverride" when multi inheritance is involved

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5640]
**Describe the bug**
See code below.

`C2` and `C3` clearly show that `f(self, x: int | str, /) -> None` is valid when overriding both `f(self, x: int, /) -> None` and `f(self, x: str, /) -> None`, but `C4` still gets reported.

**Code or Screenshots**
```python
# pyright: strict

class C0:
    def f(self, x: int, /) -> None: ...

class C1:
    def f(self, x: str, /) -> None: ...

class C2(C0):  # pass
    def f(self, x: int | str, /) -> None: ...

class C3(C1):  # pass
    def f(self, x: int | str, /) -> None: ...

class C4(C0, C1):  # reported
    def f(self, x: int | str, /) -> None: ...
```

![image](https://github.com/microsoft/pyright/assets/15389744/5e3cee71-b440-4f2d-81e0-db6cc1ca1d75)

**VS Code extension or command-line**
Pylance v2023.8.11 Pre-Release
Pyright v1.1.320

EDIT
----
Fix screenshot so that the code shown is identical to the sample code.
Fix Pylance version.

**Comment by erictraut** [comment]
This was related to [this bug report](https://github.com/python/mypy/issues/15790).

I'm going to revert the change since there appears to be a disagreement about whether this should be considered an error.

**Comment by erictraut** [comment]
The error reported in the second code sample is not related to the first one. In the second example, the error can be avoided if you change the implementation signature to:

```python
def cached_property(
    getter: Callable[[], _Value_t] | Callable[[_Owner_t], _Value_t],
) -> (
    _CachedStaticPropertyProtocol[_Value_t]
    | _CachedInstancePropertyProtocol[_Owner_t, _Value_t]
):
    raise NotImplementedError
```

**Comment by Azureblade3808** [comment]
> The error reported in the second code sample is not related to the first one. In the second example, the error can be avoided if you change the implementation signature to:
> 
> ```python
> def cached_property(
>     getter: Callable[[], _Value_t] | Callable[[_Owner_t], _Value_t],
> ) -> (
>     _CachedStaticPropertyProtocol[_Value_t]
>     | _CachedInstancePropertyProtocol[_Owner_t, _Value_t]
> ):
>     raise NotImplementedError
> ```

Thanks. I'll try it.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.321, which I just published. It will also be included in a future release of pylance.
